projects/observability/src/shared/components/timeline-card-list/timeline-card-list.component.ts

```diff
@@ -30,4 +55,40 @@ export class TimelineCardListComponent {
   public readonly dateFormat: DateFormatOptions = {
     mode: DateFormatMode.TimeWithSeconds
   };
+
+  public items: TimelineListItem[] = [];
+  public allCards: TimelineCardContainerComponent[] = [];
+
+  public ngAfterContentInit(): void {
+    queryListAndChanges$(this.cards)
+      .pipe(map(list => list.toArray()))
+      .subscribe(allCards => {
+        this.allCards = allCards;
+        this.buildItems(allCards);
+      });
+  }
+
+  public readonly isSelectedCard = (card: TimelineCardContainerComponent): boolean =>
```

Comment:
why make this an instance rather than prototype function? Intentional?

Reply:
Oh. I was trying to use htcMemoize for this, but i think it only takes one argument, which is card here. It doesn't trigger the function when selectedIndex change. If there is a way to pass more than one argument, then I can use it for with both card and selectedIndex and have it trigger only when one of them change.

Ofcourse after trying, I forgot to remove it as instance.

Reply:
Do you think it is easy to have memoize support multiple arguments?

Reply:
already does - example from memoize tests:
```typescript
    const mockFn = jest.fn((...args: string[]) => args.join(','));
    const spectator = createHost(
      `
    <pipe-holder>
      {{ hostFn | htcMemoize : arg1 : arg2 }}
    </pipe-holder>`,
      {
        hostProps: {
          hostFn: mockFn,
          arg1: 'foo',
          arg2: 'alpha'
        }
      }
    );
    expect(mockFn).toHaveBeenCalledTimes(1);
    expect(spectator.query(byText('foo,alpha'))).toExist();
```

Reply:
It does seem to support it. I think I was using a wrong syntax.

Reply:
I was passing wrong syntax. Fixed and pushing now

Reply:
thanks